libs/platform/kernel/registry_helper_km.c

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) Microsoft Corporation
+// SPDX-License-Identifier: MIT
+
+#include "ebpf_registry_helper.h"
+
+void
+close_registry_key(ebpf_registry_key_t key)
+{
+    ZwClose(key);
+}
+
+_Success_(return == 0) NTSTATUS
+    convert_guid_to_string(_In_ const GUID* guid, _Out_writes_all_(string_size) wchar_t* string, size_t string_size)
+{
+    UNICODE_STRING unicode_string;
+    NTSTATUS status = RtlStringFromGUID(guid, &unicode_string);
+    if (status != STATUS_SUCCESS) {
+        goto Exit;
+    }
+
+    if (string_size < GUID_STRING_LENGTH + 1) {
+        status = STATUS_BUFFER_TOO_SMALL;
+        goto Exit;
+    }
+
+    __analysis_assume(unicode_string.MaximumLength >= GUID_STRING_LENGTH * 2);
+    __analysis_assume(unicode_string.Buffer != NULL);
+    // Copy the buffer to the output string.
+    memcpy(string, unicode_string.Buffer, GUID_STRING_LENGTH * 2);
+    string[GUID_STRING_LENGTH] = L'\0';
+
```

Comment:
We need to call ```RtlFreeUnicodeString()``` here as well, else we leak the buffer allocated to ```unicode_string```.

Reply:
Right, same as above.